### PR TITLE
Make visualizer height flexible

### PR DESCRIPTION
**Pull request by tomneo2004**
[Trello task](https://trello.com/c/EA0R5xXT/103-fix-visualization-bug)

### To see visualizer
- Go to localhost:3000/app

### What have fixed
- Flexible width of visualizer and p5 canvas(Base on it's ancestor element)
- Flexible height of visualizer and p5 canvas(Base on it's ancestor element)
- PlayerBar is 100% screen wide at any screen width.
- Center visualizer in and p5 canvas element

### What files has changed
- `Visualizer.component.jsx`
- `Visualizer.module.scss`
- `App.module.scss`(Change from `.page-container` to `.pageContainer`)
- `sketch.js`

### Issue I have encountered
- Value of height return from visualizer's parent element is incorrect when component mounted first time but return correctly after window resized. First time calculation of p5 canvas height become value of 0 which is not the value I expect and this cause p5 library throw an error.

### Solution
- In `.visualizer` set position to absolute

### Compact screen size
- Visualizer in compact screen size is responsive but canvas get squeezed. Might consider design on how we are going to present canvas in compact screen.

### Note
- Visualizer is filled in it's ancestor element with margin and p5 canvas is always at center of visualizer because `.visualizer` content is flexbox.
- Reason to calculate width of p5 canvas is to fix PlayerBar is not 100% wide under certain width of screen.
- Reason to calculate height of p5 canvas is to let height of canvas flexible at any screen height.
- In `App.module.scss` `pageContainer` content is flexbox.

### CODE REVIEW

File: src/components/Visualizer/Visualizer.component.jsx [hunk 1]
```diff
@@ -72,8 +72,26 @@ class Visualizer extends React.Component {
         paddingWidth  - thisPaddingWidth -  
         borderWidth  -  thisBorderWidth -
         marginWidth - thisMarginWidth;
+
+        // //get padding, border and margin height for <div className={classes.visualizer}></div>
```
Comment (ArnasDickus) [inline src/components/Visualizer/Visualizer.component.jsx]: Could you explain, what this does, and why did you comment it out.

File: src/pages/App/App.module.scss [hunk 2]
```diff
@@ -1,17 +1,21 @@
 .visualmusic {
     display: flex;
     flex-direction: column;
-    padding-bottom: 64px;
+    // padding-bottom: 64px;
```
Comment (ArnasDickus) [inline src/pages/App/App.module.scss]: This is line will be removed on [PR #39](https://github.com/zero-to-mastery/visual-music/pull/39). So you can either remove it or leave it.

**Review by ArnasDickus — APPROVED** [review]
Hello, thanks for adding PR.
So looks good to me, the only thing that's bugging me, Is that every time you resize visualizer is restarting. However, I doubt you can do anything about it.

So I will try to approve this PR as soon as I get your response.

File: src/components/Visualizer/Visualizer.component.jsx (hunk 1 above)
Comment (tomneo2004) [reply src/components/Visualizer/Visualizer.component.jsx]: I have made a change and updated PR.

That part code which was comment out is to calculate height of canvas when window resize.

Regarding to visualizer is restarting when window resize. It is p5 canvas rendering , I have to look into p5 library to see whether it is possible to allow it to stop re-render when resize.

**Review by ArnasDickus — APPROVED** [review]
I ran out of concerns. Good work.

**Pull request merged**